athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchQueryUtils.java

```diff
@@ -0,0 +1,252 @@
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.elasticsearch;
+
+import com.amazonaws.athena.connector.lambda.domain.predicate.EquatableValueSet;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Range;
+import com.amazonaws.athena.connector.lambda.domain.predicate.ValueSet;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class has interfaces used for the generation of projections and predicates used for document search queries.
+ */
+class ElasticsearchQueryUtils
+{
+    private static final Logger logger = LoggerFactory.getLogger(ElasticsearchQueryUtils.class);
+
+    // Predicate conjunctions.
+    private static final String AND_OPER = " AND ";
+    private static final String OR_OPER = " OR ";
+    private static final String NO_OPER = "";
+    private static final String NOT_OPER = "NOT ";
+    private static final String LESS_THAN = "<";
+    private static final String LESS_THAN_OR_EQUAL = "<=";
+    private static final String GREATER_THAN = ">";
+    private static final String GREATER_THAN_OR_EQUAL = ">=";
+    private static final String EMPTY_PREDICATE = "";
+
+    // Existence predicates.
+    private final String existsPredicate(boolean exists, String fieldName)
+    {
+        if (exists) {
+            // (_exists:field)
+            return "(_exists_:" + fieldName + ")";
+        }
+        else {
+            // (NOT _exists_:field)
+            return "(NOT _exists_:" + fieldName + ")";
+        }
+    }
+
+    protected ElasticsearchQueryUtils() {}
+
+    /**
+     * Creates a projection (using the schema) on which fields should be included in the search index request. For
+     * complex type STRUCT, there is no need to include each individual nested field in the projection. Since the
+     * schema contains all nested fields in the STRUCT, only the name of the STRUCT field is added to the projection
+     * allowing Elasticsearch to return the entire object including all nested fields.
+     * @param schema is the schema containing the requested projection.
+     * @return a projection wrapped in a FetchSourceContext object.
+     */
+    protected FetchSourceContext getProjection(Schema schema)
+    {
+        logger.info("getProjection - enter");
+
+        List<String> includedFields = new ArrayList<>();
+
+        for (Field field : schema.getFields()) {
+            includedFields.add(field.getName());
+        }
+
+        logger.info("Included fields: " + includedFields.toString());
```

Comment:
toString() gets called automatically if you use the sl4j notation:

logger.info("Included fields: {}", includedFields);

Reply:
Removed toString()